app/src/main/java/com/zulip/android/models/Stream.java

```diff
@@ -120,9 +120,8 @@ public static Stream getByName(ZulipApp app, String name) {
             if (stream == null) {
                 Log.w("Stream.getByName",
                         "We received a stream message for a stream we don't have data for. Fake it until you make it.");
-                stream = new Stream(name);
-                app.getDao(Stream.class).createIfNotExists(stream);
-            }
+                // catch the new stream object with auto-generated id
+                stream = app.getDao(Stream.class).createIfNotExists(new Stream(name));            }
```

Comment:
I agree with the approach followed in the web client. That seems less susceptible to future bugs.